Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+pipeline {
+    agent { 
+        docker { 
+            image 'argo.registry:5000/epel-7-mgo' 
+            args '-u jenkins:jenkins'
+        }
+    }
+    options { checkoutToSubdirectory('argo-messaging') }
+    environment {
+        GOPATH="${WORKSPACE}/go"
+        GIT_COMMIT=sh(script: 'cd ${WORKSPACE}/argo-messaging && git log -1 --format="%H"',returnStdout: true).trim()
+        GIT_COMMIT_HASH=sh(script: 'cd ${WORKSPACE}/argo-messaging && git log -1 --format="%H" | cut -c1-7',returnStdout: true).trim()
+        GIT_COMMIT_DATE=sh(script: 'date -d "$(cd ${WORKSPACE}/argo-messaging && git show -s --format=%ci ${GIT_COMMIT_HASH})" "+%Y%m%d%H%M%S"',returnStdout: true).trim()
+    }
+    stages {
+        stage('Build') {
+            steps {
+                echo 'Build...'
+                sh '''
+                mkdir -p ${WORKSPACE}/go/src/github.com/ARGOeu
+                ln -sf ${WORKSPACE}/argo-messaging ${WORKSPACE}/go/src/github.com/ARGOeu/argo-messaging
+                rm -rf ${WORKSPACE}/go/src/github.com/ARGOeu/argo-messaging/argo-messaging
+                cd ${WORKSPACE}/go/src/github.com/ARGOeu/argo-messaging
+                go build
+                '''
+            }
+        }
+        stage('Test') {
+            steps {
+                echo 'Test & Coverage...'
+                sh '''
+                sudo /etc/init.d/mongod restart
+                cd ${WORKSPACE}/go/src/github.com/ARGOeu/argo-messaging
```

Comment:
I wonder if there is a patern for this path we can use in order to make this more generic e.g the git hub repo name 
@kaggis any ideas suggestions